### PR TITLE
Release 1.5.0 to develop

### PR DESCRIPTION
**Pull request by tcojean**
Add changelog, check updating third-party libraries.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,127 @@ commits. For a comprehensive list, use the following command:
...
++ Improved and optimized test random matrix generation ([#954](https://github.com/ginkgo-project/ginkgo/pull/954), [#1032](https://github.com/ginkgo-project/ginkgo/pull/1032))
++ Better CSR strategy defaults ([#969](https://github.com/ginkgo-project/ginkgo/pull/969))
++ Add `move_from` to `PolymorphicObject` ([#997](https://github.com/ginkgo-project/ginkgo/pull/997))
++ Remove unnecessary device_guard usage ([#956](https://github.com/ginkgo-project/ginkgo/pull/956))
++ Improvements to the generic accessor for mixed-precision ([#727](https://github.com/ginkgo-project/ginkgo/pull/727))
++ Add a naive lower triangular solver implementation for CUDA ([#764](https://github.com/ginkgo-project/ginkgo/pull/764))
++ Add support for int64 indices from CUDA 11 onward with SpMV and SpGEMM ([#897](https://github.com/ginkgo-project/ginkgo/pull/897))
++ Add an L1 norm implementation ([#900](https://github.com/ginkgo-project/ginkgo/pull/900))
++ Add reduce_add for arrays ([#831](https://github.com/ginkgo-project/ginkgo/pull/831))
++ Utility to simplify Dense View creation from an existing Dense vector ([#1136](https://github.com/ginkgo-project/ginkgo/pull/1136)).
```
Comment (pratikvn) [inline CHANGELOG.md:110]:
```suggestion
+ Add utility to simplify Dense View creation from an existing Dense vector ([#1136](https://github.com/ginkgo-project/ginkgo/pull/1136)).
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,127 @@ commits. For a comprehensive list, use the following command:
...
++ Adds a custom transpose implementation for Fbcsr and Csr transpose for vendor-unsupported types. ([#1123](https://github.com/ginkgo-project/ginkgo/pull/1123))
```
Comment (pratikvn) [inline CHANGELOG.md:111]:
```suggestion
+ Add a custom transpose implementation for Fbcsr and Csr transpose for unsupported vendor types. ([#1123](https://github.com/ginkgo-project/ginkgo/pull/1123))
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,127 @@ commits. For a comprehensive list, use the following command:
...
++ Deterministic IDR random init ([#1116](https://github.com/ginkgo-project/ginkgo/pull/1116))
++ Moves the algorithm choice for triangular solvers from Csr::strategy_type to a factory parameter. ([#1088](https://github.com/ginkgo-project/ginkgo/pull/1088))
```
Comment (pratikvn) [inline CHANGELOG.md:113]:
```suggestion
+ Move the algorithm choice for triangular solvers from Csr::strategy_type to a factory parameter. ([#1088](https://github.com/ginkgo-project/ginkgo/pull/1088))
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -6,6 +6,127 @@ commits. For a comprehensive list, use the following command:
...
++ Update CUDA archCoresPerSM ([#1175](https://github.com/ginkgo-project/ginkgo/pull/1116))
++ Add kernels for Csr sparsity pattern lookup ([#994](https://github.com/ginkgo-project/ginkgo/pull/994))
++ Differentiate between structural and numerical zeros in Ell/Sellp ([#1027](https://github.com/ginkgo-project/ginkgo/pull/1027))
++ Adds a binary IO format for matrix data ([#984](https://github.com/ginkgo-project/ginkgo/pull/984))
```
Comment (pratikvn) [inline CHANGELOG.md:117]:
```suggestion
+ Add a binary IO format for matrix data ([#984](https://github.com/ginkgo-project/ginkgo/pull/984))
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -6,6 +6,127 @@ commits. For a comprehensive list, use the following command:
...
++ Add a tuple zip_iterator implementation ([#966](https://github.com/ginkgo-project/ginkgo/pull/966))
++ Simplify kernel stubs and declarations ([#888](https://github.com/ginkgo-project/ginkgo/pull/888))
++ Simplify GKO_REGISTER_OPERATION with lambdas ([#859](https://github.com/ginkgo-project/ginkgo/pull/859))
++ Simplify copy to device in tests and examples ([#863](https://github.com/ginkgo-project/ginkgo/pull/863))
++ More verbose output to array assertions ([#858](https://github.com/ginkgo-project/ginkgo/pull/858))
++ Parallel Jacobi compilation ([#871](https://github.com/ginkgo-project/ginkgo/pull/871))
```
Comment (pratikvn) [inline CHANGELOG.md:123]:
```suggestion
+ Allow parallel compilation for Jacobi kernels ([#871](https://github.com/ginkgo-project/ginkgo/pull/871))
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -6,6 +6,127 @@ commits. For a comprehensive list, use the following command:
...
++ Clang-format pointer alignment change to Left ([#872](https://github.com/ginkgo-project/ginkgo/pull/872))
```
Comment (pratikvn) [inline CHANGELOG.md:124]:
```suggestion
+ Change clang-format pointer alignment to left ([#872](https://github.com/ginkgo-project/ginkgo/pull/872))
```

**Review by pratikvn — APPROVED** [review]
Some minor and optional suggestions. LGTM!

File: CHANGELOG.md [hunk 7]
```diff
@@ -6,6 +6,127 @@ commits. For a comprehensive list, use the following command:
...
++ Porting the remaining algorithms (preconditioners like ISAI, Jacobi, Multigrid, ParILU(T) and ParIC(T)) to DPC++/SYCL, update to SYCL 2020, and improve support and performance ([#896](https://github.com/ginkgo-project/ginkgo/pull/896), [#924](https://github.com/ginkgo-project/ginkgo/pull/924), [#928](https://github.com/ginkgo-project/ginkgo/pull/928), [#929](https://github.com/ginkgo-project/ginkgo/pull/929), [#933](https://github.com/ginkgo-project/ginkgo/pull/933), [#943](https://github.com/ginkgo-project/ginkgo/pull/943), [#960](https://github.com/ginkgo-project/ginkgo/pull/960), [#1057](https://github.com/ginkgo-project/ginkgo/pull/1057), [#1110](https://github.com/ginkgo-project/ginkgo/pull/1110),  [#1142](https://github.com/ginkgo-project/ginkgo/pull/1142)).
```
Comment (MarcelKoch) [inline CHANGELOG.md:47]:
```suggestion
+ Porting the remaining algorithms (preconditioners like ISAI, Jacobi, Multigrid, ParILU(T) and ParIC(T)) to DPC++/SYCL, update to SYCL 2020, and improve support and performance ([#896](https://github.com/ginkgo-project/ginkgo/pull/896), [#924](https://github.com/ginkgo-project/ginkgo/pull/924), [#928](https://github.com/ginkgo-project/ginkgo/pull/928), [#929](https://github.com/ginkgo-project/ginkgo/pull/929), [#933](https://github.com/ginkgo-project/ginkgo/pull/933), [#943](https://github.com/ginkgo-project/ginkgo/pull/943), [#960](https://github.com/ginkgo-project/ginkgo/pull/960), [#1057](https://github.com/ginkgo-project/ginkgo/pull/1057), [#1110](https://github.com/ginkgo-project/ginkgo/pull/1110),  [#1142](https://github.com/ginkgo-project/ginkgo/pull/1142))
```

**Review by MarcelKoch — APPROVED** [review]
Some minor nits.

File: third_party/rapidjson/CMakeLists.txt [hunk 8]
```diff
@@ -3,7 +3,7 @@ include(FetchContent)
 FetchContent_Declare(
     rapidjson
     GIT_REPOSITORY https://github.com/Tencent/rapidjson.git
-    GIT_TAG        48fbd8cd202ca54031fe799db2ad44ffa8e77c13
+    GIT_TAG        27c3a8dc0e2c9218fe94986d249a12b5ed838f1d
```
Comment (yhmtsai) [inline third_party/rapidjson/CMakeLists.txt:6]: I guess you pick the last one which passed their ci?

**Review by yhmtsai — APPROVED** [review]

File: CHANGELOG.md [hunk 9]
```diff
@@ -6,6 +6,127 @@ commits. For a comprehensive list, use the following command:
...
 ```
 
+
+## Version 1.5.0
+
+The Ginkgo team is proud to announce the new Ginkgo minor release 1.5.0. This release brings many important new features such as:
+- MPI-based multi-node support for all matrix formats and most solvers;
+- full DPC++/SYCL support,
+- functionality and interface for GPU-resident sparse direct solvers,
+- an inerface for wrapping solvers with automatic scaling and reordering applied,
```
Comment (upsj) [inline CHANGELOG.md:16]:
```suggestion
- an interface for wrapping solvers with scaling and reordering applied,
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -6,6 +6,127 @@ commits. For a comprehensive list, use the following command:
...
+- a new algebraic Multigrid solver,
```
Comment (upsj) [inline CHANGELOG.md:17]:
```suggestion
- a new algebraic Multigrid solver/preconditioner,
```

**Review by upsj — COMMENTED** [review]
one typo and two suggestions :)

File: third_party/rapidjson/CMakeLists.txt (hunk 8 above)
Comment (tcojean) [reply third_party/rapidjson/CMakeLists.txt:6]: yes, and since it works in all our pipelines, I guess it's fine

File: CHANGELOG.md (hunk 9 above)
Comment (upsj) [reply CHANGELOG.md:16]: I'm not sure automatic is the right adjective to describe this behavior, since it needs to be called explicitly.

**Pull request merged**